MewsSystems/mews-flutter
# feat(optimus)!: [RND-103480] Add inline dialog [ignore-size]

#### Summary

RND-103480

Added a new type of dialog - `OptimusInlineDialog`. Inline dialog is anchored to the widget and is displayed without title and small size only. Interacting outside the dialog will close it. This widget reuses a lot of the components with the default `OptimusDialog`, those parts are now moved outside and are shared between them.

- **[BREAKING CHANGE]** - `NonModalWrapper` is now `DialogWrapper` as it now let's user display `NonModalDialog` and `OptimuInlineDialog`.

<details><summary>Screenshots</summary>
<img width="468" alt="CleanShot 2022-09-08 at 10 55 14@2x" src="https://user-images.githubusercontent.com/9210422/189088833-34f4d167-9fb0-46cf-832f-c2e7b117fc0e.png">
</details> 

#### Testing steps

1. Open Storybook and test that dialog is bounded to the widget.
2. Widget should close on the action button or on tap outside the dialog.

#### Follow-up issues

None

#### Check during review

- Verify against YouTrack issue.
- Is the PR over 300 additions? Consider rejecting it with advice to split it. Is it over 500 additions? It should definitely be rejected.
- Unused code removed.
- Build passing.
- Is it a bug fix? Check that it is covered by a proper test (unit or integration).
